### PR TITLE
Add constant id to postcalendar_categories table

### PR DESCRIPTION
**Pull request by matrix-amiel**
Hi.
Here I add new column with constant_id to postcalendar_categories table.
This is missing in this table ('id' is auto increment and 'name' can changed ) and it's very necessary for a lot of reports and modules we want to add to system.

Thanks
Amiel

### CODE REVIEW

**Comment by bradymiller** [comment]
hi @matrix-amiel , 
Agree that would be good to have a hook on these that aren't tied down to the id or the title.
How does this get populated when a user adds a new category at Administration->Other->Calendar->Categories?

**Comment by matrix-amiel** [comment]
Done.

File: sql/5_0_0-to-5_0_1_upgrade.sql [hunk 1]
```diff
@@ -661,3 +661,8 @@ DELETE FROM list_options WHERE list_id = 'lists' AND option_id = 'lbfnames';
 DELETE FROM list_options WHERE list_id = 'transactions';
 DELETE FROM list_options WHERE list_id = 'lists' AND option_id = 'transactions';
 #EndIf
+
+#IfMissingColumn openemr_postcalendar_categories pc_constant_id
+ALTER TABLE `openemr_postcalendar_categories` ADD `pc_constant_id` VARCHAR (255) NOT NULL AFTER `pc_catid`;
```
Comment (bradymiller) [inline sql/5_0_0-to-5_0_1_upgrade.sql]: Better to not use the `AFTER pc_catid` .

**Comment by bradymiller** [comment]
Code looks good. Just 1 minor thing to fix that I stated above. If this is testing well for you, then let me know, and I'll bring it into the codebase (after above minor issue is addressed).

**Comment by matrix-amiel** [comment]
The changes was tested and it's working well.

File: sql/5_0_0-to-5_0_1_upgrade.sql [hunk 2]
```diff
@@ -661,3 +661,8 @@ DELETE FROM list_options WHERE list_id = 'lists' AND option_id = 'lbfnames';
 DELETE FROM list_options WHERE list_id = 'transactions';
 DELETE FROM list_options WHERE list_id = 'lists' AND option_id = 'transactions';
 #EndIf
+
+#IfMissingColumn openemr_postcalendar_categories pc_constant_id
+ALTER TABLE `openemr_postcalendar_categories` ADD `pc_constant_id` VARCHAR (255) NOT NULL;
```
Comment (bradymiller) [inline sql/5_0_0-to-5_0_1_upgrade.sql]: this should be `VARCHAR (255) default NULL;`
(to match entry in database.sql)

Comment (bradymiller) [reply sql/5_0_0-to-5_0_1_upgrade.sql]: I'll fix this issue :)

**Comment by bradymiller** [comment]
Looks good. Thanks for the improvement! -brady

**Pull request merged**